### PR TITLE
Fix a warning and cleanup a utility file

### PR DESCRIPTION
**Pull request by benson31**
Pinging @szaman19.

We had (at least) three wrappers around accumulating a vector of dimensions to produce the linear size. I think now there's just one, and I think I've caught all the equivalent replacements. There are some cases in which conversions to larger integer types are used (e.g., `accumulate(int_dims.begin(), int_dims.end(), 1UL, multiplies<size_t>())`). I left those alone for now. It's probably appropriate to change `get_linear_size` to return a `size_t` since it gives a larger buffer to protect against overflow and sizes shouldn't be signed. Perhaps I'll do that later.

### CODE REVIEW

File: src/layers/learning/distconv/distconv_layers.cpp [hunk 1]
```diff
@@ -51,8 +51,8 @@ namespace distconv{
     const auto& input_dims = input.get_local_shape();
     const auto& output_dims = output.get_local_shape();
 
-    const auto& input_size = std::accumulate(input_dims.begin(), input_dims.begin()+1, 1, std::multiplies<size_t>());
```
Comment (bvanessen) [inline src/layers/learning/distconv/distconv_layers.cpp:54]: @szaman19 I think that this is trying to get `input_dims[0] * input_dims[1]`

File: src/layers/learning/distconv/distconv_layers.cpp [hunk 2]
```diff
@@ -94,23 +105,31 @@ namespace distconv{
...
+  ::apply_bias(const tensor::Tensor<DataType, tensor::LocaleMPI, Allocator> &bias,
                  tensor::Tensor<DataType, tensor::LocaleMPI, Allocator> &output)
-  { 
+  {
     if (output.get_local_size() == 0) return 0;
 
     const auto& one = El::TypeTraits<DataType>::One();
 
     const auto& output_dims = output.get_local_shape();
-    const auto& output_size = std::accumulate(output_dims.begin(), output_dims.begin()+1, 1, std::multiplies<size_t>());
```
Comment (bvanessen) [inline src/layers/learning/distconv/distconv_layers.cpp:105]: @szaman19 I think that this is trying to get `output_dims[0] * output_dims[1]`

**Review by bvanessen — CHANGES_REQUESTED** [review]
@szaman19 Please take a look.

File: src/layers/learning/distconv/distconv_layers.cpp [hunk 3]
```diff
@@ -51,8 +51,8 @@ namespace distconv{
     const auto& input_dims = input.get_local_shape();
     const auto& output_dims = output.get_local_shape();
 
-    const auto& input_size = std::accumulate(input_dims.begin(), input_dims.begin()+1, 1, std::multiplies<size_t>());
-    const auto& output_size = std::accumulate(output_dims.begin(), output_dims.begin()+1, 1, std::multiplies<size_t>());
+    const auto& input_size = input_dims[0];
```
Comment (bvanessen) [inline src/layers/learning/distconv/distconv_layers.cpp:54]:
```suggestion
    const auto& input_size = input_dims[0] * input_dims[1];
```

File: src/layers/learning/distconv/distconv_layers.cpp [hunk 4]
```diff
@@ -94,23 +105,31 @@ namespace distconv{
...
                  tensor::Tensor<DataType, tensor::LocaleMPI, Allocator> &output)
-  { 
+  {
     if (output.get_local_size() == 0) return 0;
 
     const auto& one = El::TypeTraits<DataType>::One();
 
     const auto& output_dims = output.get_local_shape();
-    const auto& output_size = std::accumulate(output_dims.begin(), output_dims.begin()+1, 1, std::multiplies<size_t>());
+    const auto& output_size = output_dims[0];
```
Comment (bvanessen) [inline src/layers/learning/distconv/distconv_layers.cpp:116]:
```suggestion
    const auto& output_size = output_dims[0] * output_dims[1];
```

**Review by bvanessen — CHANGES_REQUESTED** [review]

File: src/layers/learning/distconv/distconv_layers.cpp [hunk 5]
```diff
@@ -180,26 +210,38 @@ namespace distconv{
...
       return 0;
     }
 
     const auto& input_dims = input.get_local_shape();
     const auto& output_dims = output_grad.get_local_shape();
 
-    const auto& input_size = std::accumulate(input_dims.begin(), input_dims.begin()+1, 1, std::multiplies<size_t>());
-    const auto& output_size = std::accumulate(output_dims.begin(), output_dims.begin()+1, 1, std::multiplies<size_t>());
+    const auto& input_size = input_dims[0];
+    const auto& output_size = output_dims[0];
```
Comment (bvanessen) [inline src/layers/learning/distconv/distconv_layers.cpp:221]:
```suggestion
    const auto& output_size = output_dims[0] * output_dims[1];
```

File: src/layers/learning/distconv/distconv_layers.cpp [hunk 6]
```diff
@@ -233,16 +275,24 @@ namespace distconv{
 
     const auto& one = El::TypeTraits<DataType>::One();
     const auto& output_dims = output_grad.get_local_shape();
-    const auto& output_size = std::accumulate(output_dims.begin(), output_dims.begin()+1, 1, std::multiplies<size_t>());
+    const auto& output_size = output_dims[0];
```
Comment (bvanessen) [inline src/layers/learning/distconv/distconv_layers.cpp:278]:
```suggestion
    const auto& output_size = output_dims[0] * output_dims[1];
```

File: src/layers/learning/distconv/distconv_layers.cpp [hunk 7]
```diff
@@ -140,18 +159,29 @@ namespace distconv{
     const auto& input_dims = input_grad.get_local_shape();
     const auto& output_dims = output_grad.get_local_shape();
 
-    const auto& input_size = std::accumulate(input_dims.begin(), input_dims.begin()+1, 1, std::multiplies<size_t>());
-    const auto& output_size = std::accumulate(output_dims.begin(), output_dims.begin()+1, 1, std::multiplies<size_t>());
+    const auto& input_size = input_dims[0];
```
Comment (bvanessen) [inline src/layers/learning/distconv/distconv_layers.cpp:162]:
```suggestion
    const auto& input_size = input_dims[0] * input_dims[1];
```

File: src/layers/learning/distconv/distconv_layers.cpp [hunk 8]
```diff
@@ -140,18 +159,29 @@ namespace distconv{
...
+    const auto& output_size = output_dims[0];
```
Comment (bvanessen) [inline src/layers/learning/distconv/distconv_layers.cpp:163]:
```suggestion
    const auto& output_size = output_dims[0] * ouput_dims[1];
```

File: src/layers/learning/distconv/distconv_layers.cpp [hunk 9]
```diff
@@ -51,8 +51,8 @@ namespace distconv{
     const auto& input_dims = input.get_local_shape();
     const auto& output_dims = output.get_local_shape();
 
-    const auto& input_size = std::accumulate(input_dims.begin(), input_dims.begin()+1, 1, std::multiplies<size_t>());
-    const auto& output_size = std::accumulate(output_dims.begin(), output_dims.begin()+1, 1, std::multiplies<size_t>());
+    const auto& input_size = input_dims[0];
+    const auto& output_size = output_dims[0];
```
Comment (bvanessen) [inline src/layers/learning/distconv/distconv_layers.cpp:55]:
```suggestion
    const auto& output_size = output_dims[0] * output_dims[1];
```

File: src/layers/learning/distconv/distconv_layers.cpp (hunk 1 above)
Comment (szaman19) [reply src/layers/learning/distconv/distconv_layers.cpp:54]: Yes, that's correct. 

We usually expect input_dims[1] == 1. The extra dimension is just needed to make the layer work nicely with distconv's channel-wise distribution. The unit test assumes that case as well.

**Review by bvanessen — APPROVED** [review]

**Pull request merged**